tests/workflow_contracts/sccache_cache_entry_test.py

```
@@ -0,0 +1,238 @@
+    The server reads SCCACHE_DIR when it starts. Unpacking the archive
+    underneath a running server leaves the run compiling from scratch while
+    the statistics report a full cache.
+    """
+    install_at = step_index(definition, lambda s: INSTALLER in run_script(s))
+    restore_at = step_index(definition, is_sccache_cache_step)
+    start_at = step_index(definition, lambda s: "--zero-stats" in run_script(s))
+    assert restore_at is not None, (
+        f"{workflow_name}:{job_name} names sccache as its wrapper with no "
+        "cache step for its directory, so nothing survives the runner"
+    )
+    assert install_at is not None and start_at is not None
```

Ruff (0.16.3)

[warning] Assertion should be broken down into multiple parts

Break down assertion into multiple parts

(PT018)